app/models/my_module.rb

```diff
@@ -225,6 +225,37 @@ def live_and_snapshot_repositories_list
     (live_repositories + selected_snapshots).sort_by { |r| r.name.downcase }
   end
 
+  def active_snapshot_or_live(rep_or_snap_id, exclude_snpashot_ids: [])
+    r = RepositoryBase.find_by(id: rep_or_snap_id)
+    return unless r
+
+    parent_id = r.is_a?(Repository) ? r.id : r.parent_id
+
+    selected_snapshot_for_repo(parent_id, exclude_snpashot_ids: exclude_snpashot_ids) ||
+      assigned_repositories&.where(id: parent_id)&.first ||
+      repository_snapshots
+        .where(parent_id: parent_id)
+        .where.not(id: exclude_snpashot_ids)
+        .order(updated_at: :desc).first
+  end
+
+  def update_report_references(rep_or_snap_id)
+    rep_or_snap = RepositoryBase.find(rep_or_snap_id)
+
+    ids = if rep_or_snap.is_a?(Repository)
+            RepositorySnapshot.where(parent_id: rep_or_snap.id).pluck(:id)
+          else
+            Repository.where(id: rep_or_snap.parent_id).pluck(:id) +
+              RepositorySnapshot.where(parent_id: rep_or_snap.parent_id).pluck(:id)
+          end
+
+    report_elements.where(repository_id: ids).update(repository_id: rep_or_snap.id)
+  end
+
+  def selected_snapshot_for_repo(repository_id, exclude_snpashot_ids: [])
+    repository_snapshots.where(parent_id: repository_id).where.not(id: exclude_snpashot_ids).selected.first
```

Comment:
Just add 'selected' condition to where. There is no much sense to have scope for this, as long as there can be only one selected snapshot.